File: 278. First Bad Version/278. First Bad Version.rb
# The is_bad_version API is already defined for you.
# @param {Integer} version
# @return {boolean} whether the version is bad
# def is_bad_version(version):

# @param {Integer} n
# @return {Integer}
def first_bad_version(n)
    head = 1
    tail = n
    while head < tail
        mid = Integer((head + tail) / 2) # change to integer
        if is_bad_version(mid)
            tail = mid
        else
            head = mid + 1
        end
    end
    return head
end

File: 278. First Bad Version/278. First Bad Version.c
// Forward declaration of isBadVersion API.
bool isBadVersion(int version);

int firstBadVersion(int n) {
    int head = 1, tail = n, mid;
    while (head < tail){
        mid = head + (tail - head) / 2; // force to int = int + float
        if (isBadVersion(mid))
            tail = mid;
        else
            head = mid + 1;
    }
    return head;
}
